### DISCUSSION TITLE
VSCode syntax highlighting

### DISCUSSION THREAD

**Discussion by blackerby** [johnkerl/miller#1138] [Show and tell]
Greetings, all,

First, @johnkerl thank you for this marvelous tool.

I spent the day porting the Vim syntax file that ships with Miller to a tmLanguage.json file for use with VSCode. It's my first VSCode extension, so the code quality may be a bit rough around the edges, but it seems to work (that is, it at least passes the tests in [testfile.mlr](https://github.com/johnkerl/miller/blob/main/vim/syntax/testfile.mlr)).

The extension is available on the [VSCode Marketplace](https://marketplace.visualstudio.com/items?itemName=WilliamBlackerby.miller-dsl) and the repo is [here](https://github.com/blackerby/vscode-miller-dsl-syntax)

**Comment by aborruso** [comment]
thank you very much

**Comment by blackerby** [comment]
My pleasure. I abandoned an attempt at doing the same for Emacs [here](https://github.com/blackerby/miller-mode.el) a couple of weeks ago. VSCode was much easier for a hobbyist like me.
